### PR TITLE
Simple homepage and general refactoring

### LINKED ISSUE #130: Create Dashboard home page
**Tasks**:

- [x] Fix redirects
- [x] Set correct Navbar buttons
- [x] Add title
- [x] Add brief description

### PR DESCRIPTION
**Pull request by cedoor**
## Description

This PR adds a simple homepage, replace the old Reach hooks with simple utility functions (as there wasn't any React state) and replaces the old routes with the recommended router (i.e. `createBrowserRouter`), which enables [loaders](https://reactrouter.com/en/main/route/loader), [actions](https://reactrouter.com/en/main/route/action), [fetchers](https://reactrouter.com/en/main/hooks/use-fetcher) and more.

## Related Issue

Closes #130 

## Does this introduce a breaking change?

-   [ ] Yes
-   [x] No

### CODE REVIEW

File: apps/dashboard/src/main.tsx [hunk 1]
```diff
@@ -42,15 +47,60 @@ const wagmiClient = createClient({
...
+            const { pathname } = new URL(request.url)
+
+            if (
+                (pathname === "/" || pathname === "/sso") &&
+                (await isLoggedIn())
+            ) {
+                throw redirect("/my-groups")
+            }
+
+            return null
```
Comment (saleel) [inline apps/dashboard/src/main.tsx:64]: Oh, that is a nice trick I didn't know of.

File: apps/dashboard/src/pages/home.tsx [hunk 2]
```diff
@@ -1,5 +1,39 @@
...
+                        <Heading as="h2" size="2xl" textAlign="center">
+                            Welcome to ZKGroups
+                        </Heading>
+                        <Text fontSize="xl">
+                            A system for managing privacy-preserving groups.
+                        </Text>
+
+                        <Link href="/sso">
+                            <Button colorScheme="primary" variant="solid">
+                                Launch app
```
Comment (saleel) [inline apps/dashboard/src/pages/home.tsx:31]: 🙏

File: apps/dashboard/src/utils/semaphoreAPI.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
+import { Subgraph } from "@semaphore-protocol/subgraph"
```
Comment (saleel) [inline apps/dashboard/src/utils/semaphoreAPI.ts:1]: Nice 👍 
This model is definitely better than having hooks for everything where all it does is export stateless functions. We can put these two APis under a `data` folder though as they are not `utils`? 🤔 
Anyway, we would need another refactoring here when we introduce SIWE and have same account create/display both on-chain and off-chain groups. The components wont have to worry about which flow it is (on-chain v off-chain) and can call same function to create/get both group types.

**Review by saleel — APPROVED** [review]
Great PR 🚀 
Added some minor comments

File: apps/dashboard/src/components/create-group-modal.tsx [hunk 4]
```diff
@@ -58,7 +57,7 @@ export default function CreateGroupModal({
         }
     }, [_groupSize, nextStep])
 
-    async function createGroup(
+    async function createOnchainGroup(
```
[the remark this answers is not in the document]
Comment (cedoor) [reply apps/dashboard/src/components/create-group-modal.tsx]: Ops, right!

File: apps/dashboard/src/utils/semaphoreAPI.ts (hunk 3 above)
Comment (cedoor) [reply apps/dashboard/src/utils/semaphoreAPI.ts:1]: I renamed that folder to `api`, as they both are basically API. How does it sound?

Ah right, we'll update them 👍🏽

Comment (saleel) [reply apps/dashboard/src/utils/semaphoreAPI.ts]: Make sense 👍

**Review by saleel — APPROVED** [review]

**Pull request merged**